state/types.go

```diff
@@ -1,28 +1,45 @@
 package state
 
 import (
+	"math/big"
+
+	"github.com/0xPolygonHermez/zkevm-node/pool"
+
 	"github.com/0xPolygonHermez/zkevm-node/state/runtime/instrumentation"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 )
 
+type TouchedAddress struct {
+	Address common.Address
+	Nonce   *uint64
+	Balance *big.Int
+}
+
+// ProcessBatchRequest represents the request of a batch process.
+type ProcessBatchRequest struct {
+	BatchNumber      uint64
+	StateRoot        common.Hash
+	GlobalExitRoot   common.Hash
+	OldAccInputHash  common.Hash
+	TxData           []byte
+	SequencerAddress common.Address
+	Timestamp        uint64
+	IsFirstTx        bool
```

Comment:
I think this is not needed here. This should be used for the GER empty hack, but we're already taking into consideration this in the finalizer.

With that being said, since this is something related to how the executor works, it will make a lot of sense to take care bout this using the `isFirstTx`, but then we don't need to do it in the finalizer. Also if it's the state who needs to do that, then it also needs to know if the inteeractions is meant to be done in the `tx by tx` mode or in the `full batch mode`...

Up to you, essentially you have to either:

- Remove `IsFirstTx` from the struct
- Remoe the GER = empty hash from finalizer, and add a `tx by tx` flag in the process request struct

Reply:
Okay